retailer-oas/retailer.yaml

```diff
@@ -499,6 +719,196 @@ paths:
+                        - 'customer is unavailable'
+                        - 'customer requested to cancel'
+                        - 'shopper unable to complete order'
+                        - 'shopper could not find address'
+                        - 'too many out of stock items'
+                        - 'canceled by retailer'
+                        - 'fraudulent customer'
+                        - 'store early closure'
+                        - 'system error'
+                        - 'other'
+                        - 'unknown'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OrderCancelation'
+                required: ['id', 'status', 'createdAt', 'order']
+        '400':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    description: Error message
+                  errorCode:
+                    type: string
+                    description: Error code
+                    enum:
+                      - "ALREADY_CANCELED"
+                      - "TOO_LATE_TO_CANCEL"
+                      - "INVALID_REASON"
+                  errors:
+                    type: array
+                    items:
+                      type: string
+                      description: List of error details
+        '404':
+          description: Order Not Found
+  
+    
+  /webhooks/subscriptions/event-types:
+    get:
+      summary: List webhook event types
+      description: Endpoint to see all the event types that you can subscribe to.
+      operationId: listWebhookEventTypes
+      tags:
+        - webhooks
+      parameters: []
+      responses:
+        '200':
+          description: OK
+          
+  # TODO: add this to test subscription
+  # /webhooks/subscriptions/{id}/test:
+  #   post:
+  #     summary: Test webhook subscription
+  #     description: Tests a webhook subscription by sending a test event to the notification URL.
+  #     operationId: testWebhookSubscription
+  #     tags:
+  #       - webhooks
+  #     parameters: []
+  #     responses:
+  #       '200':
+  #         description: OK
+
+
+  /webhooks/subscriptions:
+    get:
+      summary: List webhook subscriptions
+      description: Lists all webhook subscriptions owned by an application.
+      operationId: listWebhookSubscription
+      tags:
+        - webhooks
+      parameters: []
+      responses:
+        '200':
+          description: OK
+    post:
+      summary: Create new webhook subscription
+      description: |-
+        Subscribe to receive events. 
+        Use this endpoint to add your notification URL and list of events that you want your application to subscribe to.
+      operationId: createWebhookSubscription
```

Comment:
Should we receive an URL where we will send the request later on? how they can ensure that we are the ones doing the requests, those requests are signed?